test/integration/dra/binding_conditions_test.go

```diff
@@ -0,0 +1,609 @@
+	maxTO := wantTO * 19 / 10 // 1.9 * wantTO
+
+	// Start the scheduler with a short binding timeout.
+	cfg := fmt.Sprintf(`
+profiles:
+- schedulerName: default-scheduler
+  pluginConfig:
+  - name: DynamicResources
+    args:
+      bindingTimeout: %s
+`, wantTO)
+
+	startSchedulerWithConfig(tCtx, cfg)
+
+	// Create claim+pod: allocation happens, then scheduler waits in PreBind.
+	claim1 := createClaim(tCtx, namespace, "-timeout-enforced", class, claim)
+	pod := createPod(tCtx, namespace, "-timeout-enforced", podWithClaimName, claim1)
+
+	// Wait until the claim is allocated.
+	allocatedClaim := waitForClaimAllocatedToDevice(tCtx, namespace, claim1.Name, 10*time.Second)
+
+	gomega.NewWithT(tCtx).Expect(allocatedClaim).To(gomega.HaveField(
+		"Status.Allocation",
+		gstruct.PointTo(gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+			"Devices": gomega.Equal(resourceapi.DeviceAllocationResult{
+				Results: []resourceapi.DeviceRequestAllocationResult{{
+					Request:                  allocatedClaim.Spec.Devices.Requests[0].Name,
+					Driver:                   driver,
+					Pool:                     poolWithBinding,
+					Device:                   "with-binding",
+					BindingConditions:        []string{bindingCondition},
+					BindingFailureConditions: []string{failureCondition},
+				}},
+			}),
+		}),
+		)), "Claim must be allocated to the condition-gated device")
+
+	tStart := time.Now()
+
+	// The scheduler should hit the binding timeout and surface that on the pod.
+	// We poll the pod's conditions until we see a message containing "binding timeout".
+	ktesting.Eventually(tCtx, func(tCtx ktesting.TContext) *v1.Pod {
+		p, err := tCtx.Client().CoreV1().Pods(namespace).Get(tCtx, pod.Name, metav1.GetOptions{})
+		tCtx.ExpectNoError(err, "get pod")
+		return p
+	}).WithTimeout(maxTO).WithPolling(300*time.Millisecond).Should(
+		gomega.HaveField("Status.Conditions",
+			gomega.ContainElement(
+				gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+					"Message": gomega.ContainSubstring("binding timeout"),
+				}),
+			),
+		),
+		"pod should report binding timeout in a condition message",
+	)
+
+	elapsed := time.Since(tStart)
+	gomega.NewWithT(tCtx).Expect(elapsed).To(
+		gomega.BeNumerically("<=", maxTO),
+		"bindingTimeout should trigger roughly near %s (observed %v)", wantTO, elapsed,
+	)
+	// Verify that the pod remains unscheduled after the binding timeout.
+	ktesting.Eventually(tCtx, func(tCtx ktesting.TContext) *v1.Pod {
+		pod, err := tCtx.Client().CoreV1().Pods(namespace).Get(tCtx, pod.Name, metav1.GetOptions{})
+		tCtx.ExpectNoError(err, "get pod")
+		return pod
+	}).WithTimeout(wantTO).WithPolling(200 * time.Millisecond).Should(gomega.SatisfyAll(
+		gomega.HaveField("Spec.NodeName", gomega.BeEmpty()),
+
+		gomega.WithTransform(func(p *v1.Pod) []v1.PodCondition {
+			return p.Status.Conditions
+		}, gomega.Not(gomega.ContainElement(v1.PodCondition{
+			Type:   v1.PodScheduled,
+			Status: v1.ConditionTrue,
+		}))),
```

Comment:
My suggestion was to replace WithTransform with HaveField in general, so this also becomes:


```suggestion
		gomega.HaveField("Status.Conditions", gomega.Not(gomega.ContainElement(v1.PodCondition{
			Type:   v1.PodScheduled,
			Status: v1.ConditionTrue,
		}))),
```